docs/source/resources/graphql-glossary.md

```diff
@@ -0,0 +1,240 @@
+
+<h2 id="resolver">Resolver</h2>
+<p>A function that connects schema fields and types to various backends. It can retrieve or write data from either an SQL, a No-SQL, graph database, a micro-service or a REST API.</p>
+
+
+<h2 id="schema">Schema</h2>
+<p>A model of the data that can be fetched from or written to a GraphQL server.</p>
+
+
+<h2 id="schema-definition-language">Schema Definition Language (SDL)</h2>
+<p>The syntax for writing GraphQL Schemas. It is otherwise known as Interface Definition Language. It is the lingua franca shared by all for building GraphQL APIs regardless of the programming language chosen.</p>
+
+
+<h2 id="schema-first-development">Schema first development</h2>
+<p>A development approach for designing and building modern UIs that involves the frontend and backend teams agreeing on a Schema first, which serves as a contract between the UI and the backend before any API engineering happens.</p>
+
+
+<h2 id="schema-registry">Schema registry</h2>
+<p>A central database that enables schema registration, tracking of detailed schema changes e.g. types added, fields added, fields deprecated and checking out previous versions of schema.</p>
+
+
+<h2 id="schema-versioning">Schema versioning</h2>
+<p>Refers to the ability to have different versions of your Schema. This allows for reversible changes to be made to the Schema. The Apollo CLI is a tool that provides and manages Schema versioning with Engine.</p>
+
+
+<h2 id="schema-stitching">Schema stitching</h2>
+<p>The process of merging [different schemas into one GraphQL schema](./docs/graphql-tools/schema-stitching.html). These schemas can be local, remote or from third party services.</p>
+
+
+<h2 id="subscription">Subscription</h2>
+<p>A real-time GraphQL operation. A Subscription is defined in a schema like queries and mutations.</p>
+
+```js
+type Subscription {
+  commentAdded(repoFullName: String!): Comment
+}
+...
+subscription onCommentAdded($repoFullName: String!){
+  commentAdded(repoFullName: $repoFullName){
+    id
+    content
+  }
+}
+```
+
+<h2 id="scalar-type">Scalar Type</h2>
+<p>A type that qualifies the data a GraphQL field resolves. GraphQL ships with some scalar types out of the box; **Int**, **Float**, **String**, **Boolean** and **ID**. However, a custom scalar type such as **Date** can be specified in a GraphQL service implementation.</p>
+
+
+<h2 id="type-system">Type System</h2>
+<p>A collection of types which characterizes the set of data that can be validated, queried and executed on a GraphQL API.</p>
+
+
+<h2 id="whole-response-caching">Whole response caching</h2>
+<p>A technique used to cache entire results of GraphQL queries. This process improves performance by preventing the fetching of the same results from the server if it has been obtained before. Check out the [Apollo performance guide on how to implement this type of caching](../guides/performance.html).</p>
```

Comment:
Maybe change this link to be shorter, like just around `[Apollo performance guide]`?